src/StdGas.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.2 <0.9.0;
+
+import {VmSafe} from "./Vm.sol";
+import {console} from "./console.sol";
+
+abstract contract StdGas {
+    VmSafe private constant vm = VmSafe(address(uint160(uint256(keccak256("hevm cheat code")))));
+
+    // Stores the storage slots that have been accessed before the last recording.
+    mapping(address => mapping(bytes32 => bool)) private accessed;
+
+    // Gas left when the metering started.
+    uint256 gasLeftAtStart;
+
+    function startGasMetering() internal {
+        // Store gas left before recording.
+        gasLeftAtStart = gasleft();
+        // Clear accesses list.
+        vm.record();
```

Comment:
:warning: I think I don't understand how this works and what is recorded

Reply:
Seen with @MathisGD: the `vm.accesses` function returns all the accesses (read & write) since the last `vm.record`